Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/PersonExtra.hs

```diff
@@ -365,32 +366,42 @@ updatePersonRec (Id personId) person = do
     ]
     [Se.Is BeamP.id (Se.Eq personId)]
 
-updatePersonVersions :: (MonadFlow m, EsqDBFlow m r, CacheFlow m r) => Person -> Maybe Version -> Maybe Version -> Maybe Version -> Maybe Text -> Maybe Text -> m ()
-updatePersonVersions person mbBundleVersion mbClientVersion mbConfigVersion mbDevice' mbBackendApp = do
+updatePersonVersions ::
+  (MonadFlow m, EsqDBFlow m r, CacheFlow m r) =>
+  Person ->
+  Maybe Version ->
+  Maybe Version ->
+  Maybe Version ->
+  Maybe Text ->
+  Maybe Text ->
+  Id DMOC.MerchantOperatingCity ->
+  m ()
+updatePersonVersions person mbBundleVersion mbClientVersion mbConfigVersion mbDevice' mbBackendApp city = do
   let mbDevice = getDeviceFromText mbDevice'
-  when
-    ((isJust mbBundleVersion || isJust mbClientVersion || isJust mbDevice' || isJust mbConfigVersion) && (person.clientBundleVersion /= mbBundleVersion || person.clientSdkVersion /= mbClientVersion || person.clientConfigVersion /= mbConfigVersion || person.clientDevice /= mbDevice || person.backendAppVersion /= mbBackendApp))
-    do
-      now <- getCurrentTime
-      let mbBundleVersionText = versionToText <$> (mbBundleVersion <|> person.clientBundleVersion)
-          mbClientVersionText = versionToText <$> (mbClientVersion <|> person.clientSdkVersion)
-          mbConfigVersionText = versionToText <$> (mbConfigVersion <|> person.clientConfigVersion)
-          mbOsVersion = deviceVersion <$> (mbDevice <|> person.clientDevice)
-          mbOsType = deviceType <$> (mbDevice <|> person.clientDevice)
-          mbModelName = deviceModel <$> (mbDevice <|> person.clientDevice)
-          mbManufacturer = deviceManufacturer =<< (mbDevice <|> person.clientDevice)
-      updateOneWithKV
-        [ Se.Set BeamP.clientSdkVersion mbClientVersionText,
-          Se.Set BeamP.clientBundleVersion mbBundleVersionText,
-          Se.Set BeamP.clientConfigVersion mbConfigVersionText,
-          Se.Set BeamP.clientOsVersion mbOsVersion,
-          Se.Set BeamP.clientOsType mbOsType,
-          Se.Set BeamP.clientModelName mbModelName,
-          Se.Set BeamP.clientManufacturer mbManufacturer,
-          Se.Set BeamP.backendAppVersion mbBackendApp,
-          Se.Set BeamP.updatedAt now
-        ]
-        [Se.Is BeamP.id (Se.Eq $ getId person.id)]
+  let isBundleDataPresent = isJust mbBundleVersion || isJust mbClientVersion || isJust mbDevice' || isJust mbConfigVersion
+  let isAnyMismatchPresent = or [person.clientBundleVersion /= mbBundleVersion, person.clientSdkVersion /= mbClientVersion, person.clientConfigVersion /= mbConfigVersion, person.clientDevice /= mbDevice, person.backendAppVersion /= mbBackendApp, person.merchantOperatingCityId /= city]
+  when (isBundleDataPresent && isAnyMismatchPresent) $ do
```

Comment:
What if isAnyMismatchPresent is true because of merchantOperatingCityId but isBundleDataPresent false?

Reply:
it will be always present.

Reply:
in auth it always comes from ui.